### ISSUE TITLE
Deselecting a Factory While using Multi-Unit Hotkeys Breaks Left-Click

### ISSUE THREAD

**Issue by misterperson** [FAForever/fa#3169]
Repro steps:
Start a standard game
W/ ACU Build a Land Factory
while factory is building:
- Queue T2 HQ upgrade
- Press engineer alternate hotbuild hotkey
- While unit selection window is open, use the "Select All Builders" hotkey or something similar to select the ACU 
Left-click no longer works.

Right-clicking w/ ACU selected works around the left-click issue, BUT
whenever a factory is selected and an alternate hotbuild hotkey is pressed, left clicking will never deselect the factory but instead queue the last-used alternate hotbuild (even if a unique unit is then queued like mobile shield / stealth).

https://replay.faforever.com/12662993
The engineer I queue'd to repro the bug doesn't show up in the replay factory queue, either!

**Comment by shalkya** [comment]
yeah you need to quit the cycle before selecting something else. By either right clicking, or middle clicking. I didn't think about selecting other unit through hotkey when i made that, I will try to find a way to avoid this issue though (maybe prevent people from selecting other unit while you are still in building unit mod; maybe by quitting the mod when pressing another hotkey)

**Comment by lL1l1** [comment]
Can still reproduce in current version.
Need to exit build mode on selection changed.
